ICAFIX/ReApplyFixMultiRunPipeline.sh

```diff
@@ -678,94 +697,106 @@ export FSL_FIX_WBC="${Caret7_Command}"
-		log_Msg "${matlab_cmd}"
+			log_Msg "FSL_FIX_MCR: ${FSL_FIX_MCR}"
+							
+			matlab_cmd=("${matlab_exe}" "${FSL_FIX_MCR}" "${matlab_function_arguments[@]}")
+
+			# redirect tokens must be parsed by bash before doing variable expansion, and thus can't be inside a variable
+			# MPH: Going to let Compiled MATLAB use the existing stdout and stderr, rather than creating a separate log file
+			#matlab_logfile=".reapplyfixmultirun.${concatfmri}${RegString}.fix_3_clean.matlab.log"
+			#log_Msg "Run MATLAB command: ${matlab_cmd[*]} >> ${matlab_logfile} 2>&1"
+			#"${matlab_cmd[@]}" >> "${matlab_logfile}" 2>&1
+			log_Msg "Run compiled MATLAB: ${matlab_cmd[*]}"
+			"${matlab_cmd[@]}"
+			;;
 		
-		# Use bash redirection ("here-string") to pass multiple commands into matlab
-		# (Necessary to protect the semicolons that separate matlab commands, which would otherwise
-		# get interpreted as separating different bash shell commands)
-		(
-			#fix's default settings.sh isn't safe to unbound variables or exit codes, so disable everything before sourcing
-			debug_disable_trap
-			set +u
-			source "${FSL_FIXDIR}/settings.sh"
-			set -u
-			debug_enable_trap
-			export FSL_FIX_WBC="${Caret7_Command}"
-			"${interpreter[@]}" <<<"${matlab_cmd}"
-		)
-		;;
-
-	*)
-		# Unsupported MATLAB run mode
-		log_Err_Abort "Unsupported MATLAB run mode value: ${MatlabRunMode}"
-		;;
-esac
+		1 | 2)
+			# Use interpreted MATLAB or Octave
+			if [[ ${MatlabRunMode} == "1" ]]; then
+				interpreter=(matlab -nojvm -nodisplay -nosplash)
+			else
+				interpreter=(octave-cli -q --no-window-system)
+			fi
 
-log_Msg "Done running fix_3_clean"
+			if (( DoVol )); then
+				matlab_cmd="${ML_PATHS} fix_3_clean('${fixlist}',${aggressive},${MotionRegression},${AlreadyHP});"
+			else
+				matlab_cmd="${ML_PATHS} fix_3_clean('${fixlist}',${aggressive},${MotionRegression},${AlreadyHP},${DoVol});"
+			fi
+			
+			log_Msg "Run interpreted MATLAB/Octave (${interpreter[@]}) with command..."
+			log_Msg "${matlab_cmd}"
+			
+			# Use bash redirection ("here-string") to pass multiple commands into matlab
+			# (Necessary to protect the semicolons that separate matlab commands, which would otherwise
+			# get interpreted as separating different bash shell commands)
+			(
+				#fix's default settings.sh isn't safe to unbound variables or exit codes, so disable everything before sourcing
+				debug_disable_trap
+				set +u
+				source "${FSL_FIXDIR}/settings.sh"
+				set -u
+				debug_enable_trap
+				export FSL_FIX_WBC="${Caret7_Command}"
+				"${interpreter[@]}" <<<"${matlab_cmd}"
+			)
+			;;
+
+		*)
+			# Unsupported MATLAB run mode
+			log_Err_Abort "Unsupported MATLAB run mode value: ${MatlabRunMode}"
+			;;
+	esac
+
+	log_Msg "Done running fix_3_clean"
 
+else
+	log_Msg "Running fix -a"
+	fix_cmd="${FSL_FIXDIR}/fix -a ${fixlist}"
+	if ((MotionRegression));then
+		fix_cmd="${fix_cmd} -m -h ${AlreadyHP}"
+	fi
+	if ((aggressive));then
+		fix_cmd="${fix_cmd} -A"
+	fi
+	${fix_cmd}
```

Comment:
Arrays are built for this, and give whitespace safety.
```suggestion
	fix_cmd=("${FSL_FIXDIR}"/fix -a "${fixlist}")
	if ((MotionRegression));then
		fix_cmd+=(-m -h "${AlreadyHP}")
	fi
	if ((aggressive));then
		fix_cmd+=(-A)
	fi
	"${fix_cmd[@]}"
```